packages/theme-addresses/README.md

```diff
@@ -0,0 +1,133 @@
+# @shopify/theme-addresses
+
+theme-addresses.js helps developers work with common address releated operations, such as, building the country and province selectors
+
+## Getting Started
+
+Theme Scripts can be used in any theme project. To take advantage of semantic versioning and easy updates, we recommend using NPM or Yarn to include them in your project:
+
+```
+yarn add @shopify/theme-addresses
+```
+
+and then import the functions you wish to use through ES6 imports:
+
+```
+import {CountryProvinceSelector} from '@shopify/theme-addresses`;
+```
+
+If you prefer not to use a package manager, you can download the latest version of Theme Product and include it in your project manually from the following links:
+
+- [theme-addresses.js](http://unpkg.com/@shopify/theme-addresses@latest/dist/theme-addresses.js)
+- [theme-addresses.min.js](http://unpkg.com/@shopify/theme-addresses@latest/dist/theme-addresses.min.js)
+
+## Methods
+
+### CountryProvinceSelector(countryOptions)
+
+Prepares an instance of a `CountryProvinceSelector` with the given country options
+
+- `countryOptions`: This is an html string generated by the following liquid tags:
+    * [`{{ all_country_option_tags | json }}`](https://help.shopify.com/en/themes/liquid/objects/all-country-option-tags)
+    * [`{{ country_option_tags | json }}`](https://help.shopify.com/en/themes/liquid/objects/country-option-tags)
+
+Example
+```html
+// In theme.liquid, somewhere before the end of </head> tag
+    ...
+    {% if template.directory == 'customers' %}
+      theme.allCountryOptionTags = {{ all_country_option_tags | json }};
+    {% endif %}
+  </script>
+  ...
+</head>
+```
+```js
+// In addresses.js
+import {CountryProvinceSelector} from '@shopify/theme-addresses';
+
+const billingCountryProvinceSelector = new CountryProvinceSelector(window.theme.allCountryOptionTags);
+```
+
+---
+
+### build(countryNodeElement, provinceNodeElement, options)
+
+Populates the given `<select>` elements with country and province options.
+
+- `countryNodeElement`: The `<select>` node element for countries
+- `provinceNodeElement`: The `<select>` node element for provinces according to the selected country
+- `options`:
+    * `onCountryChange`: The callback after a country `change` event
+    * `onProvinceChange`: The callback after a province `change` event
+
+The Select node element should look like the following:
+```html
+    <select ... id="newCountry" data-default="{{ form.country }}"></select>
+    <select ... id="newProvince" data-default="{{ form.province }}"></select>
```

Comment:
Let's include `name="address[province]"` since its also required for the form to submit properly.